### PR TITLE
Specific cap-add and cap-drop command

### PR DESCRIPTION
**Pull request by q384566678**
The management of each field of the specific Capabilities.

Signed-off-by: zhouhao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

**Comment by Mashimiao** [comment]
need rebase

**Comment by q384566678** [comment]
updated.

File: cmd/oci-runtime-tool/generate.go [hunk 1]
```diff
@@ -77,7 +77,17 @@ var generateFlags = []cli.Flag{
 	cli.StringFlag{Name: "output", Usage: "output file (defaults to stdout)"},
 	cli.BoolFlag{Name: "privileged", Usage: "enable privileged container settings"},
 	cli.StringSliceFlag{Name: "process-cap-add", Usage: "add Linux capabilities"},
```
Comment (Mashimiao) [inline cmd/oci-runtime-tool/generate.go]: process-cap-add, process-cap drop should be removed

File: generate/generate.go [hunk 2]
```diff
@@ -923,6 +923,101 @@ func (g *Generator) AddProcessCapability(c string) error {
...
 }
 
+// AddProcessAmbientCapability adds a process capability into g.spec.Process.Capabilities.Ambient.
+func (g *Generator) AddProcessAmbientCapability(c string) error {
+	cp := strings.ToUpper(c)
+	if err := validate.CapValid(cp, g.HostSpecific); err != nil {
+		return err
+	}
+
+	g.initSpec()
```
Comment (Mashimiao) [inline generate/generate.go]: should be g.initSpecProcessCapabilities()

File: man/oci-runtime-tool-generate.1.md [hunk 3]
```diff
@@ -277,9 +277,39 @@ read the configuration from `config.json`.
 **--process-cap-add**=[]
   Add Linux capabilities
 
+**--process-cap-add-ambient**=[]
```
Comment (Mashimiao) [inline man/oci-runtime-tool-generate.1.md]: Don't forget to modify completion file

**Comment by q384566678** [comment]
updated, PTAL.

**Comment by q384566678** [comment]
@Mashimiao @liangchenye @hqhq PTAL

File: generate/generate.go [hunk 4]
```diff
@@ -883,8 +883,27 @@ func (g *Generator) ClearProcessCapabilities() {
 	g.spec.Process.Capabilities.Ambient = []string{}
 }
 
-// AddProcessCapability adds a process capability into g.spec.Process.Capabilities.
-func (g *Generator) AddProcessCapability(c string) error {
+// AddProcessAmbientCapability adds a process capability into g.spec.Process.Capabilities.Ambient.
+func (g *Generator) AddProcessAmbientCapability(c string) error {
```
Comment (Mashimiao) [inline generate/generate.go]: As we did for other generator functions, function's name should be based on spec's item order. As setting value for g.spec.Process.Capabilities.Ambient, the name should be AddProcessCapabilitiesAmbient.

Comment (q384566678) [inline generate/generate.go]: Done.

**Comment by Mashimiao** [comment]
LGTM

**Comment by q384566678** [comment]
ping @mrunalp @liangchenye @hqhq PTAL

**Comment by q384566678** [comment]
ping @liangchenye

**Comment by q384566678** [comment]
reping @opencontainers/runtime-tools-maintainers

**Comment by q384566678** [comment]
ping @liangchenye @mrunalp

**Comment by q384566678** [comment]
rebased @Mashimiao @hqhq @liangchenye @mrunalp PTAL

**Comment by Mashimiao** [comment]
LGTM

**Comment by liangchenye** [comment]
LGTM

**Pull request merged**